mods/ctf/ctf_combat/ctf_healing/bandage.lua

```diff
@@ -9,72 +9,83 @@ function ctf_healing.register_bandage(name, def)
-			local hp_add = math.random(def.heal_min or 3, def.heal_max or 4)
-
-			if hp + hp_add > limit then
-				hp_add = limit - hp
-				hp = limit
-			else
-				hp = hp + hp_add
-			end
-
-			local result = RunCallbacks(ctf_healing.registered_on_heal, player, object, hp_add)
-
-			if not result then
-				object:set_hp(hp)
-				hud_events.new(pname, {
-					quick = true,
-					text = uname .. " healed you!",
-					color = 0xC1FF44,
-				})
-			elseif type(result) == "string" then
-				hud_events.new(uname, {
-					quick = true,
-					text = result,
-					color = "warning",
-				})
-			end
-		end,
+			if pointed_thing.type == "object" then
+
+                local object = pointed_thing.ref
+			    if not object:is_player() then return end
+
+			    local pname = object:get_player_name()
+			    local uname = player:get_player_name()
+
+			    if pname == uname then return end
+
+			    if ctf_teams.get(pname) ~= ctf_teams.get(uname) then
+				    hud_events.new(uname, {
+					    quick = true,
+					    text = pname .. " isn't in your team!",
+					    color = "warning",
+				    })
+				    return
+			    end
+
+			    local hp = object:get_hp()
+			    local limit = def.heal_percent * object:get_properties().hp_max
+
+			    if hp <= 0 then
+				    hud_events.new(uname, {
+					    quick = true,
+					    text = pname .. " is dead!",
+					    color = "warning",
+				    })
+				    return
+			    end
+
+			    if hp >= limit then
+				    hud_events.new(uname, {
+					    quick = true,
+					    text = pname .. " already has " .. limit .. " HP!",
+					    color = "warning",
+				    })
+				    return
+			    end
+
+			    local hp_add = math.random(def.heal_min or 3, def.heal_max or 4)
+
+				if hp + hp_add > limit then
+				    hp_add = limit - hp
+				    hp = limit
+			    else
+				    hp = hp + hp_add
+			    end
+
+			    local result = RunCallbacks(ctf_healing.registered_on_heal, player, object, hp_add)
+
+			    if not result then
+				    object:set_hp(hp)
+				    hud_events.new(pname, {
+					    quick = true,
+					    text = uname .. " healed you!",
+					    color = 0xC1FF44,
+				    })
+			    elseif type(result) == "string" then
+				    hud_events.new(uname, {
+					    quick = true,
+					    text = result,
+					    color = "warning",
+				    })
+                end
+	        elseif pointed_thing.type == "node" then
+                local node_pointed = minetest.get_node(pointed_thing.under)
+                local node_above = minetest.get_node(pointed_thing.under:offset(0, 1, 0))
+                if node_pointed.name ~= "ctf_modebase:flag_captured_top" then
+                    if node_pointed.name:find("ctf_modebase:flag_") then
+		                ctf_modebase.flag_on_punch(player, pointed_thing.under, node_pointed)
+                    elseif node_above.name:find("ctf_modebase:flag_") then
+                        ctf_modebase.flag_on_punch(player, pointed_thing.under:offset(0, 1, 0), node_above)
+                    end
```

Comment:
```suggestion
                local node_pointed = minetest.get_node(pointed_thing.under)
                local node_above = minetest.get_node(pointed_thing.under:offset(0, 1, 0))
                local flag_captured_top = "ctf_modebase:flag_captured_top"
                if node_pointed.name ~= "ctf_modebase:flag_captured_top" then
                    if node_pointed.name:find("ctf_modebase:flag_") then
		                ctf_modebase.flag_on_punch(player, pointed_thing.under, node_pointed)
                    elseif node_above.name:find("ctf_modebase:flag_") and node_above.name ~= flag_captured_top then
                        ctf_modebase.flag_on_punch(player, pointed_thing.under:offset(0, 1, 0), node_above)
                    end
```

It was crashing when the flag was taken and the bottom was clicked with medic's bandage. This fixes it.